### PR TITLE
fix achievements page

### PR DESCRIPTION
**Pull request by notewio**
@ahsu1230

### CODE REVIEW

**Comment by ahsu1230** [comment]
Looks like you need to run the code formatter: Prettier.
In gemini-user, run this:
```
npm install prettier -g                         <- you may have to run this as administrator
npm run start
```

**Comment by notewio** [comment]
the goal is to recreate the [current achievements page](https://www.andymathnavigator.com/#/student-achievements), so it sorts the list of achievements by year and then displays them accordingly

File: constellations/gemini-user/src/achievements/achievements.sass [hunk 1]
```diff
@@ -1,2 +1,61 @@
...
+
+
+.achieve-card
+	margin-bottom: 96px
+	
+	& h2
+		line-height: 30px
+		font-size: 18px
+		color: #056571
+		border-bottom: 1px solid #222
```
Comment (mythicat554) [inline constellations/gemini-user/src/achievements/achievements.sass:12]: It's important to properly order CSS, so take a look at this file: https://github.com/ahsu1230/mathnavigatorSite/blob/master/resources/frontend/css_ordering.md
So border-bottom would come first, then color, then font-size and line-height.

File: constellations/gemini-user/src/achievements/achievements.sass [hunk 2]
```diff
@@ -1,2 +1,61 @@
...
+
+	& li
+		height: 96px
+		line-height: 24px
+		border-bottom: 1px solid #b8b8b8
+		font-size: 16px
+		
+		display: flex
+		flex-direction: row
+		align-items: center
```
Comment (mythicat554) [inline constellations/gemini-user/src/achievements/achievements.sass:22]: Same thing here with the ordering; positioning should be first, then display, then color and text.

File: constellations/gemini-user/src/achievements/achievements.sass [hunk 3]
```diff
@@ -1,2 +1,61 @@
...
+		margin-top: 96px
+		line-height: 28px
+		text-align: center
+	
+	& .subheaders
+		margin: 32px 0 96px
+		line-height: 26px
+		font-size: 18px
+		color: #222
+		text-align: center
```
Comment (mythicat554) [inline constellations/gemini-user/src/achievements/achievements.sass:58]: Group font-size, line-height, and text-align together below color

File: constellations/gemini-user/src/achievements/achievements.js [hunk 4]
```diff
@@ -17,11 +40,40 @@ export class AchievementPage extends React.Component {
...
+        let sorted = {};
+        achievements.forEach((achievement) => {
+            if (!(achievement.year in sorted)) {
+                sorted[achievement.year] = [];
+            }
+            sorted[achievement.year].push(achievement);
+        });
+
+        let years = Object.keys(sorted);
+        years.sort().reverse();
```
Comment (ahsu1230) [inline constellations/gemini-user/src/achievements/achievements.js:56]: I've actually had problems with `Object.keys`, especially in Internet Explorer.
Look around the codebase for examples where we use `import {...} from lodash;` to see how we use the `lodash` library to make our lives easier. You can find documentation here: https://lodash.com/docs/4.17.15

In this case, I'd probably look at the `_.keys` function

File: constellations/gemini-user/src/achievements/achievements.js [hunk 5]
```diff
@@ -17,11 +40,40 @@ export class AchievementPage extends React.Component {
...
     render() {
+        const achievements = this.state.achieveList;
+
+        let sorted = {};
+        achievements.forEach((achievement) => {
+            if (!(achievement.year in sorted)) {
+                sorted[achievement.year] = [];
+            }
+            sorted[achievement.year].push(achievement);
+        });
```
Comment (ahsu1230) [inline constellations/gemini-user/src/achievements/achievements.js:53]: I think you do this easily using `_.groupBy` from lodash: https://lodash.com/docs/4.17.15#groupBy

File: constellations/gemini-user/src/achievements/achievements.sass [hunk 6]
```diff
@@ -1,2 +1,61 @@
...
+
+	
+
```
Comment (ahsu1230) [inline constellations/gemini-user/src/achievements/achievements.sass:61]: too many spaces at the end of this file

File: constellations/gemini-user/src/achievements/achievements.js [hunk 7]
```diff
@@ -18,10 +42,31 @@ export class AchievementPage extends React.Component {
     }
 
     render() {
+        const achievements = this.state.achieveList;
+
+        let sorted = groupBy(achievements, (a) => a.year);
+
+        let years = sortBy(keys(sorted)).reverse();
+
```
Comment (ahsu1230) [inline constellations/gemini-user/src/achievements/achievements.js:50]: nit: too many blank lines here. You can just condense them

```
const achievements =
let sorted = 
let years = 
```

File: constellations/gemini-user/src/achievements/achievements.sass [hunk 8]
```diff
@@ -1,2 +1,57 @@
 #view-achieve h1
-  color: red
+	color: red
+
+
+.achieve-card
```
Comment (ahsu1230) [inline constellations/gemini-user/src/achievements/achievements.sass:5]: This file's tab spaces are a bit weird.
Can you make them 4-space tabs please? We'll try to make all files size 4 tabs as a standard.

**Comment by ahsu1230** [comment]
Looks good to me! Go ahead and merge!

**Pull request merged**